client/ayon_core/addon/utils.py

```diff
@@ -150,35 +148,52 @@ def ensure_addons_are_process_context_ready(
         detail = output_str
 
     _handle_error(process_context, message, detail)
-    if not exit_on_failure:
-        return exception
-    sys.exit(1)
+    if exit_on_failure:
+        sys.exit(1)
 
 
 def ensure_addons_are_process_ready(
+    addon_name: Optional[str] = None,
+    addon_version: Optional[str] = None,
+    project_name: Optional[str] = None,
+    headless: Optional[bool] = None,
     addons_manager: Optional[AddonsManager] = None,
     exit_on_failure: bool = True,
     **kwargs,
-) -> Optional[Exception]:
+) -> ProcessContext:
     """Ensure all enabled addons are ready to be used in the given context.
```

Comment:
So with the arguments now exposed (which I like) there is something now that strikes me as odd. Why does something that tells me whether "all addons are ready" take an addon name and version as input argument?

Is this doing it for all addons, or just the one? With the arguments now exposed it sounds more like this is more a `ensure_addon_is_process_ready` function?

Or should we improve the docstring to elaborate the use case? Why does it take an input addon if it checks for ALL addons to be ready?

Reply:
https://github.com/ynput/ayon-core/pull/849/files#diff-6f20bc5b8745075bff6ad36db99c83009e00f864cfff5cefee4f56905c16d846R171

```
addon_name (Optional[str]): Addon name which triggered process.
addon_version (Optional[str]): Addon version which triggered process.
```

Reply:
I see - so one addon, requests other addons?

Reply:
Well, technically one addon requests ayon-core to prepare stuff, and one of steps in ayon-core is that it asks addons to prepare them (if they have the logic implemented).